src/Storages/ObjectStorageQueue/ObjectStorageQueueSource.cpp

```diff
@@ -648,15 +648,9 @@ void ObjectStorageQueueSource::commit(bool success, const std::string & exceptio
 
 void ObjectStorageQueueSource::applyActionAfterProcessing(const String & path)
 {
-    switch (queue_settings.after_processing.value)
+    if (files_metadata->getTableMetadata().after_processing == "delete")
```

Comment:
I do not understand why we need string instead of enum/simple bool value here but if there are any reasons, just skip this comment

Reply:
Because in table metadata it was always a string and now we have to use TableMetadata everywhere. I actually made it a enum in a separate PR https://github.com/ClickHouse/ClickHouse/pull/70811

Reply:
Ok, agree, no need to change this immediately